kitchen.do.yml

```diff
@@ -0,0 +1,84 @@
+---
+driver:
+  name: digitalocean
+  size: s-1vcpu-1gb
+  region: nyc3
+
+provisioner:
+  name: puppet_apply
+  ignore_spec_fixtures: true
+  manifests_path: test/fixtures/manifests.do
+  modules_path: test/fixtures/modules
+  puppetfile_path: test/fixtures/Puppetfile
+  require_chef_for_busser: false
+
+platforms:
+  - name: centos-6-puppet5
+    driver_config:
+      image: centos-6-x64
+    provisioner:
+      puppet_yum_repo: https://yum.puppetlabs.com/puppetlabs-release-el-6.noarch.rpm
```

Comment:
There is way to avoid duplications of this with vagrant and DO setup, kitchen.yml is processed by ERB before it gets evaluated as yaml, so following tricks are possible with it (example from my private chef monorepo):

```yaml
---
<% if (ENV['DIGITALOCEAN_ACCESS_TOKEN'] && ENV['DIGITALOCEAN_SSH_KEY_IDS'] && ENV['SSH_KEY_PATH']) || ENV['CI'] %>
driver:
  name: digitalocean
  size: s-1vcpu-1gb
  region: fra1
  ipv6: true

transport:
  ssh_key: <%= Monorepo.path(ENV['SSH_KEY_PATH']) %>
  max_wait_until_ready: 30
<% else %>
driver:
  name: vagrant
<% end %>

<% if (ENV['DIGITALOCEAN_ACCESS_TOKEN'] && ENV['DIGITALOCEAN_SSH_KEY_IDS'] && ENV['SSH_KEY_PATH']) || ENV['CI'] || $KITCHEN_FORCE_CI_PLATFORMS %>
platforms:
- name: ubuntu-16.04
  driver_config:
    image: ubuntu-16-04-x64
- name: centos-7
  driver_config:
    image: centos-7-x64
- name: debian-9
  driver_config:
    image: debian-9-x64
<% else %>
platforms:
  - name: ubuntu-16.04
    driver_config:
      box: bento/ubuntu-16.04
      gui: false
  - name: centos-7
    driver_config:
      box: bento/centos-7.5
  - name: debian-9
    driver_config:
      box: bento/debian-9.6
  - name: mint-18-cinnamon
    driver_config:
      box: artem-sidorenko/mint-18.3-cinnamon
      gui: false
      customize:
        memory: 1024
        vram: 32
<% end %>

provisioner: &global-provisioner
  name: chef_zero
  data_bags_path: <%= Monorepo.path('test/fixtures/data_bags') %>
  roles_path: <%= Monorepo.path('roles') %>
  attributes:
    kitchen-test-run: true
    kitchen-fast-test: <%= (ENV['CI_COMMIT_REF_SLUG'] == 'master' && (ENV['CI_PIPELINE_ID'].to_i % 3).zero?) ? 'false':'true' %>
  log_level: 'warn'
  product_name: 'chef'
  product_version: '14.12.3'
  #deprecations_as_errors: true

verifier:
  name: inspec
  attributes:
    kitchen-fast-test: <%= (ENV['CI_COMMIT_REF_SLUG'] == 'master' && (ENV['CI_PIPELINE_ID'].to_i % 3).zero?) ? 'false':'true' %>
<% if File.exist?('test/integration/attributes.yml') %>
  attrs:
    - test/integration/attributes.yml
<% end %>
```

just to give an idea :-)

Reply:
but maybe it makes sense to try this out in a dedicated PR if you like

Reply:
Wohoho ... cool way, but I also think this should be done in a separate PR